### PR TITLE
Histogram fillnd

### PR DESCRIPTION
**Pull request by jrmlhermitte**
fixed arg checking and type checking. PEP fixes as well

### CODE REVIEW

File: skxray/core/accumulators/histogram.pyx [hunk 1]
```diff
@@ -186,8 +185,8 @@ class Histogram:
         # numerical data type
         numtypes = [np.dtype(int), np.dtype(float)]
         numtypeindex = {tp : i for i, tp in enumerate(numtypes)}
-        ctypeindices = [numtypeindex.get(c.dtype, MAX_DIMENSIONS)
-                for c in coords]
+        ctypeindices = [numtypeindex.get(c.dtype, len(numtypes)+1)
```
Comment (jrmlhermitte) [inline skxray/core/accumulators/histogram.pyx:188]: I think this should be something greater than len(numtypes)-1. I put len(numtypes)+1 but forgot we have zero based indexing here from enumerate. Do you agree?

File: skxray/core/accumulators/tests/test_histogram.py [hunk 2]
```diff
@@ -69,6 +69,7 @@ def test_2d_histogram():
         [[ten, ten], xi, yi, 1],
         [[ten, nine], xf, yf, wi],
         [[ten, nine], xi, yi, wf],
+        [[ten, nine], xf, yi, wi],
```
Comment (jrmlhermitte) [inline skxray/core/accumulators/tests/test_histogram.py:72]: @pavoljuhas fillnd fails on this test. Any idea what's wrong?
(first bins are floats but second bins are integers, seems to fail on any permuation of weights)

**Comment by pavoljuhas** [comment]
Great catch!  `dataindexstrides` need to be reordered according to `coordsorder`.  I feel this part is ready for a rewrite  I don't like using an arbitrary sort key (`len(numpytes)`) for unknown numerical types.  This actually does not matter because unhandled types would raise an exception later.  I will update the histogram-fillnd branch (and #6) to make these things look cleaner.

**Comment by pavoljuhas** [comment]
Incorporated in bac0c02468eafb3fa0ae821e3862870f7d8c6bb0.  Please rebase any new PR-s from my add-histogram branch.

**Pull request closed without merging**